### PR TITLE
PYD-1399: Add environments documentation

### PR DESCRIPTION
**Pull request by ChristopherGS**
Documents the environment feature

Edit by Marcelo:
- Closes https://github.com/pydantic/logfire/issues/319

### CODE REVIEW

File: docs/guides/advanced/using-environments.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+Note that you can also set this via the `LOGFIRE_ENVIRONMENT` environment variable.
+
+Once set, you will see your environment in the Logfire UI `all envs` dropdown, which appears
+on the [Live View](../web-ui/live.md), [Dashboards](../web-ui/dashboards.md) and [Explore](../web-ui/explore.md) pages:
+
+![Environments](../../images/guide/environments.png)
+
+Note that by default there are system generated environments:
+
+- `all envs`: Searches will include all spans with any environment set
```
Comment (ChristopherGS) [inline docs/guides/advanced/using-environments.md]: @Kludex is this description of `all_envs` vs. `not specified` correct, I wasn't sure. Could it be that `not specified` just returns everything?

File: docs/guides/advanced/using-environments.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+You can set a Logfire environment via the SDK `logfire.configure` function:
+
+```py title="main.py"
+import logfire
+
+logfire.configure(environment='dev')
+
+from app.main import main
+
+main()
```
Comment (Kludex) [inline docs/guides/advanced/using-environments.md]: I prefer to have examples people can reproduce by _copy-pasting_.
```suggestion
logfire.info("Hi there!")
```

**Comment by ChristopherGS** [comment]
Not sure what's going on with the build

File: docs/guides/advanced/using-environments.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
+If you set up Logfire according to the [getting started guide](../../index.md), you already have instrumented logfire in your applications
+via the `logfire.configure()` function. If you find yourself wanting to distinguish between data sent in one environment (e.g. `dev` vs. `production`), it may be overkill to create a new project for each (more on how to decide this below).
+
+Instead you can use the environments feature, which is a special kind of resource attribute applied to the whole payload received by Logfire.
+This attribute says which environment the payload comes from.
+
+You can set a Logfire environment via the SDK `logfire.configure` function:
```
Comment (Kludex) [inline docs/guides/advanced/using-environments.md]: I wouldn't call it "Logfire environment", just "environment".

```suggestion
You can set the environment for your project when calling `logfire.configure`:
```

File: docs/guides/advanced/using-environments.md (hunk 1 above)
Comment (Kludex) [reply docs/guides/advanced/using-environments.md]: I don't know... I was not involved in this decision. 🤷‍♂️

It makes sense for `all envs` to include `not specified`, but I don't know how it was implemented.

File: docs/guides/advanced/using-environments.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+on the [Live View](../web-ui/live.md), [Dashboards](../web-ui/dashboards.md) and [Explore](../web-ui/explore.md) pages:
+
+![Environments](../../images/guide/environments.png)
+
+Note that by default there are system generated environments:
+
+- `all envs`: Searches will include all spans with any environment set
+- `not specified`: Searches will include all spans with no environment specified
+
+Any environments you create via the SDK will appear below the system generated envs. When you select an environment,
```
[the remark this answers is not in the document]
Comment (Kludex) [reply docs/guides/advanced/using-environments.md]: I think this bot is actually correct.

File: docs/guides/advanced/using-environments.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+all subsequent queries (e.g. on live view, dashboards or explore) will filter by that environment.
+
+## Can I Create an Environment in the UI?
+No, you cannot create or delete set environments via the UI, instead use the SDK.
```
Comment (Kludex) [inline docs/guides/advanced/using-environments.md]: I wouldn't include unnecessary information. 👀

File: mkdocs.yml [hunk 6]
```diff
@@ -103,6 +103,7 @@ nav:
           - Link to Code Source: guides/advanced/link-to-code-source.md
           - Testing: guides/advanced/testing.md
           - Backfill: guides/advanced/backfill.md
+          - Using Environments: guides/advanced/using-environments.md
```
Comment (Kludex) [inline mkdocs.yml]: I think this should go on the top, it's going to be a very common use case.

File: docs/guides/advanced/using-environments.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+## How do I delete an environment?
+Once an environment has been configured and received by logfire, technically it’s available for
+the length of the data retention period while that environment exists in the data.
+You can however add new ones, and change the configuration of which data is assigned to which
+environment name.
+
+## Should I Use Environments or Projects?
+
+Environments are more lightweight than projects. Projects give you the ability to assign specific
+user groups and permissions levels (see this [organization structure diagram](docs/reference/organization-structure/) for details).
```
Comment (Kludex) [inline docs/guides/advanced/using-environments.md]: Can you use relative path to the docs like the other links, please?

File: docs/guides/advanced/using-environments.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+
+logfire.configure(environment='dev')
+
+from app.main import main
+
+main()
+```
+
+Under the hood, this sets the OTel [`deployment.environment.name`](https://opentelemetry.io/docs/specs/semconv/resource/deployment-environment/).
+Note that you can also set this via the `LOGFIRE_ENVIRONMENT` environment variable.
```
Comment (Kludex) [inline docs/guides/advanced/using-environments.md:14]: I think more important than `LOGFIRE_ENVIRONMENT` would be to explain how to set this in other languages... e.g. with the `OTEL_RESOURCE_ATTRIBUTES="deployment.environment.name=prod"` environment variable.

File: docs/guides/advanced/using-environments.md (hunk 1 above)
Comment (e-hosseini) [reply docs/guides/advanced/using-environments.md]: All envs has no limitations,

Comment (e-hosseini) [reply docs/guides/advanced/using-environments.md]: `not specified` returns logs with empty env value

File: docs/guides/advanced/using-environments.md (hunk 5 above)
Comment (ChristopherGS) [reply docs/guides/advanced/using-environments.md]: Disagree, I think it's worth being explicit about this so people don't try and find it. There is very little downside to clarifying it.

File: docs/guides/advanced/using-environments.md (hunk 8 above)
Comment (ChristopherGS) [reply docs/guides/advanced/using-environments.md:14]: Thanks, added

File: docs/guides/advanced/using-environments.md (hunk 7 above)
Comment (ChristopherGS) [reply docs/guides/advanced/using-environments.md]: fixed

File: mkdocs.yml (hunk 6 above)
Comment (ChristopherGS) [reply mkdocs.yml]: updated

File: docs/guides/advanced/using-environments.md [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+
+You can set the environment for your project when calling `logfire.configure`:
+
+```py title="main.py"
+import logfire
+
+logfire.configure(environment='dev')
+
+logfire.info("Hi there!")
+```
```
Comment (Kludex) [inline docs/guides/advanced/using-environments.md]: I think I need to be a bit more annoying. I think we can simplify this, and be more concise.

I don't think the first paragraph introduces well what we are going to see in this page.

```suggestion
As a developer, we find ourselves working on differents environments for a project: local, production, sometimes staging, and depending on your company deployment strategy... You can have even more! 😅 

With **Logfire** you can distinguish which environment you are sending data to.
You just need to set the the [`environment` parameter on the `logfire.configure`][logfire.configure(environment)].

    ```py title="main.py"
    import logfire

    logfire.configure(environment='local')  # (1)!
    ```
1. Usually you are retriving the environment information from an environment variable. 
```

Comment (ChristopherGS) [reply docs/guides/advanced/using-environments.md]: yeah this is better

**Pull request merged**